pages/mesosphere/dcos/services/edge-lb/1.5/concepts/edge-lb-architecture-deep-dive/index.md

```diff
@@ -0,0 +1,53 @@
+The Edge-LB API receives incoming client requests and, based on the action requested, creates, deletes, or updates Edge-LB pool instances. The API server also serves API endpoint requests for Edge-LB pool(s) instances.
+
+The Edge-LB API server uses the DC/OS ZooKeeper persistent storage to store the configuration file for the load balancer and communicates to the Edge-LB pool instances through ZooKeeper&reg; transactions. The interaction between ZooKeeper and the Edge-LB API server enables Edge-LB to be fault tolerant. In the event of an API server failure, the Edge-LB pool instances continue to load-balance traffic to the proper backend. When the Edge-LB API server comes back online, it re-establishes the communication between itself and the pool instances.
+
+The Edge-LB API server has two sub-components:
+- Apache&reg; Mesos&reg; listener
+- DC/OS template
+
+## Mesos listener
+The `mesos-listener` sub-component is always up by default. It subscribes to the Mesos event stream, translates the event stream data into an internal data structure, and serves metadata from the data structure through the API server. It also runs a gRPC server to serve the Mesos task information to other Edge-LB components like the `dcos-template`.
+
+## DC/OS template
+The `dcos-template` sub-component dynamically renders the load balancer configuration information for HAProxy&reg; by combining the `mesos-listener` output and the Edge-LB pool configuration provided by the Edge-LB CLI client. The template then signals the `lbmgr` sub-component to take appropriate action on the HAProxy load balancer instance.
+
+In the event of Edge-LB pool update, the `dcos-template` re-generates the HAProxy load balancer configuration from the Edge-LB pool configuration file and signals `lbmgr` to reload the existing HAProxy configuration in real-time.
+
+# A closer look at Edge-LB pools
+The Edge-LB pool provides the core load balancing functionality for Edge-LB software.
+However, there are no Edge-LB pool instances created by default.
+To create a pool instance, you must define configuration settings in a pool configuration file.
+You can set up load balancing for one service or for multiple services using the same pool configuration file, depending on your application and network requirements.
+
+Edge-LB pool has two sub-components:
+- Load balancing manager (lbmgr)
+- HAProxy
+
+## Load balancing manager (lbmgr)
+
+The load balancing manager (`lbmgr`) in Edge-LB pool instances is a non-distributed process that runs along with the HAProxy load balancer. `lbmgr` takes the appropriate action for the pool when it receives requests from the API server.
+
+The `lbmgr` component monitors the HAProxy instances for the Edge-LB pool and restarts HAProxy in the same container, when needed. For example, if an HAProxy instance goes down because an agent is disconnected, the `lbmgr` process creates a new HAProxy instance to replace it on another available agent.
```

Comment:
> it would be the EdgeLB server (not lbmgr) that would try to create a new instance on another agent.

Just to make already great comment from @cneth more precise, it is the SDK scheduler that restart pool server instances in case when they fail. lbmgr restarts the haproxy if needed, but it roles is limited to the single container.

Reply:
I think I understand, but probably best to check, since I am still new. Is this statement correct?:

"The `lbmgr` component monitors the HAProxy instances for the Edge-LB pool and restarts HAProxy in a single container, when needed. If an HAProxy instance goes down because an agent is disconnected, the SDK scheduler restarts the HAProxy instance on another available agent."